### ISSUE TITLE
Improve the Ignore method

### ISSUE THREAD

**Issue by bboyle1234** [dotnet/orleans#4728]
The `Ignore` extension method implemented in Orleans, copied below, won't behave as expectedwhen the task is canceled because it uses the `TaskContinuationOptions.OnlyOnFaulted` option. Canceled tasks are not considered faulted and their exception won't be observed.

Therefore, tasks that are ignored but get canceled and will still throw an unhandled `OperationCanceledException` or `TaskCanceledException`.

The fix is to either a) update the method summary noting the task will still throw task cancellation exceptions, or b) use `TaskContinuationOptions.NotOnRanToCompletion` instead.

``` c#
 /// <summary>
    /// Utility functions for dealing with Tasks.
    /// </summary>
    public static class PublicOrleansTaskExtensions
    {
        private static readonly Action<Task> IgnoreTaskContinuation = t => { var ignored = t.Exception; };

        /// <summary>
        /// Observes and ignores a potential exception on a given Task.
        /// If a Task fails and throws an exception which is never observed, it will be caught by the .NET finalizer thread.
        /// This function awaits the given task and if the exception is thrown, it observes this exception and simply ignores it.
        /// This will prevent the escalation of this exception to the .NET finalizer thread.
        /// </summary>
        /// <param name="task">The task to be ignored.</param>
        [SuppressMessage("Microsoft.Performance", "CA1804:RemoveUnusedLocals", MessageId = "ignored")]
        public static void Ignore(this Task task)
        {
            if (task.IsCompleted)
            {
                var ignored = task.Exception;
            }
            else
            {
                task.ContinueWith(
                    IgnoreTaskContinuation,
                    CancellationToken.None,
                    TaskContinuationOptions.OnlyOnFaulted | TaskContinuationOptions.ExecuteSynchronously,
                    TaskScheduler.Default);
            }
        }
    }
```

**Comment by ReubenBond** [comment]
Good fine, @bboyle1234 - could you submit a PR?

**Comment by bboyle1234** [comment]
I'm not 100% certain (need to yield to higher experience) ... but I think this test code shows that unobserved, canceled tasks, do not raise `TaskScheduler.UnobservedTaskException` events, and that I was wrong in my initial assumptions creating this Issue, and that my changes were unnecessary. 

In the following code, `TaskScheduler.UnobservedTaskException` is never triggered.

```C#
using System;
using System.Threading;
using System.Threading.Tasks;

namespace ConsoleApp3 {
    class Program {

        static void Main(string[] args) {

            TaskScheduler.UnobservedTaskException += (s, e) => {
                Console.WriteLine("Unobserved exception was thrown.");
                Console.Write(e.Exception.ToString());
                throw e.Exception; // Crashes the application - we wanna nail unobserved exceptions dead.
            };


            var tcs1 = new TaskCompletionSource<int>();
            var tcs2 = new TaskCompletionSource<int>();
            tcs1.SetCanceled();
            tcs1.Task.Ignore();
            tcs2.Task.Ignore();
            tcs2.SetCanceled();

            for (var i = 0; i < 10; i++) Thread.Sleep(100);

            Console.WriteLine("Press Enter to exit.");
            Console.ReadLine();
        }
    }

    public static class Extensions {

        public static void Ignore(this Task task) {
            if (task.IsCompleted) {
                var x = task.Exception;
                Console.WriteLine($"x: {x?.ToString() ?? "null"}");
            } else {
                task.ContinueWith(t => {
                    var x = task.Exception;
                    Console.WriteLine($"x: {x?.ToString() ?? "null"}");
                },
                CancellationToken.None,
                TaskContinuationOptions.OnlyOnFaulted | TaskContinuationOptions.ExecuteSynchronously,
                TaskScheduler.Default);
            }
        }
    }
}
```

**Comment by ReubenBond** [comment]
Fixed by #4729
